cloudinit/sources/azure/imds.py

```diff
@@ -80,6 +81,19 @@ def exception_callback(self, req_args, exception) -> bool:
         else:
             log = False
 
+        # Always log if error does not match previous.
+        if exception.code is not None:
+            if self._last_error != exception.code:
+                log = True
+                self._last_error = exception.code
+        elif (
+            self._last_error is None
+            or not isinstance(self._last_error, type)
```

Comment:
Sorry, I am trying to understand this line. Could you please explain what this checks?

Reply:
If no previous error or, or previous error is not an instance of the current one, log it/save it.

Reply:
So if say previous error was a connection timeout and the new error is 404, we'll log the 404.

Reply:
If specifically referring to `or not isinstance(self._last_error, type)`

We're checking that _last_error is a _type_ like saved below.  If it's not a type, it means the previous error was an exception with http code (int).

Reply:
Specifically it handles the case where we transition from http code error (e.g. 404) -> connection timeout or similar exception.

Reply:
Added code docs